File: Year-2023/LeetCode-April-Challenge/BoatstoSavePeople.js
var numRescueBoats = function (people, limit) {
    people.sort((a, b) => a - b);
    let low = 0, high = people.length - 1;
    let boatCount = 0;
    while(low <= high){
        if(people[low] + people[high] <= limit) low++;
        high--;
        boatCount++;
    }
    return boatCount;
};

let people = [1,2], limit = 3;
console.log(numRescueBoats(people, limit));

File: Year-2023/LeetCode-April-Challenge/BoatstoSavePeople.java
import java.util.*;
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int low = 0, high = people.length - 1;
        int boatCount = 0;
        while(low <= high){
            if(people[low] + people[high] <= limit) low++;
            high--;
            boatCount++;
        }
        return boatCount;
    }
}
public class BoatstoSavePeople{
    public static void main(String[] args) {
        int[] people = {1,2}; int limit = 3;
        Solution s = new Solution();
        System.out.println(s.numRescueBoats(people, limit));
    }
}